src/Query/QueryBuilder.php

```diff
@@ -1548,4 +1559,37 @@ public function __clone()
             $this->params[$name] = clone $param;
         }
     }
+
+    /**
+     * Enables caching of the results of this query, for given amount of seconds
+     * and optionally specified witch key to use for the cache entry.
+     *
+     * @return $this
+     */
+    public function enableResultCache(QueryCacheProfile $cacheProfile): self
+    {
+        $this->resultCacheProfile = $cacheProfile;
+
+        return $this;
+    }
+
+    /**
+     * Disables caching of the results of this query.
+     *
+     * @return $this
+     */
+    public function disableResultCache(): self
+    {
+        $this->resultCacheProfile = null;
+
+        return $this;
+    }
+
+    /**
+     * Gets the cache profile for the result caching.
+     */
+    public function getResultCacheProfile(): ?QueryCacheProfile
```

Comment:
I'm curious if we really need to introduce this method. The builder will use the result cache profile for executing the queries that it has built. Who else except for the builder is supposed to use it?

Reply:
That is a very good point. I think, we should remove the method.

Reply:
The method was successfully eliminated.